modules/mesh-layers/src/scenegraph-layer/scenegraph-layer-vertex.glsl.js

```diff
@@ -67,7 +68,12 @@ void main(void) {
     geometry.uv = pbr_vUV;
   #endif
 
-  vec3 pos = (instanceModelMatrix * (sceneModelMatrix * POSITION).xyz) * sizeScale + instanceTranslation;
+  float sizeMinPixels = sizePixels.x;
+  float sizeMaxPixels = sizePixels.y;
+  float originalSize = project_size_to_pixel(sizeScale);
+  float clampedSize = clamp(originalSize, sizeMinPixels, sizeMaxPixels);
+
+  vec3 pos = (instanceModelMatrix * (sceneModelMatrix * POSITION).xyz) * sizeScale * (clampedSize / originalSize) + instanceTranslation;
```

Comment:
Perf: `viewport`, `sizeScale` and `size*Pixels` are the same for all vertices. You should be able to get the scaling factor on CPU instead of calculating it in the vertex shader.

Reply:
Can I refactor later/in another diff?